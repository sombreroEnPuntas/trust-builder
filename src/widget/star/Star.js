import React from 'react';
import PropTypes from 'prop-types';

import './Star.scss';

const Star = ({ index, rating, size }) => (
  <span className="Star">
    <span
      className={`star-block_color-${index <= rating ? rating : 0} star-block_size-${size}`}
    >
      ðŸŸŠ
    </span>
  </span>
);

Star.propTypes = {
  size: PropTypes.oneOf(['small', 'big']).isRequired,
  index: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
};


Star.defaultProps = {
  size: 'small',
};

export default Star;
