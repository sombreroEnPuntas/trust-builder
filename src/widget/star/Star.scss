@import '../styles';

.Star {
  .star-block_size-small {
    margin: $font-size-primary-star-margin;
    height: $font-size-primary-star-block;
    width: $font-size-primary-star-block;
    font-size: $font-size-primary-star;
  }

  .star-block_size-big {
    margin: $font-size-primary-big-star-margin;
    height: $font-size-primary-big-star-block;
    width: $font-size-primary-big-star-block;
    font-size: $font-size-primary-big-star;
  }

  @for $i from 0 through 5 {
    .star-block_color-#{$i} {
      @include colored-star-blocks($i);
    }
  }
}
