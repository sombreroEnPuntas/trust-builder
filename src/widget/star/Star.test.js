import React from 'react';
import { shallow } from 'enzyme';

import Star from './Star';

it('Star renders without crashing', () => {
  shallow(<Star index={1} rating={5} />);
});

describe('Star gets proper class name and structure for:', () => {
  const stars = [
    {index: 2, rating: 1}, // color-0
    {index: 1, rating: 1}, // color-1
    {index: 1, rating: 2}, // color-2
    {index: 1, rating: 3}, // color-3
    {index: 1, rating: 4}, // color-4
    {index: 1, rating: 5}, // color-5
  ];

  stars.forEach(star => {
    it(`index ${star.index} / rating: ${star.rating}`, () => {
      const wrapper = shallow(<Star {...star} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});

describe('Star gets proper class name and structure for:', () => {
  const stars = [
    {index: 1, rating: 1, size: 'big'},
  ];

  stars.forEach(star => {
    it(`index ${star.index} / rating: ${star.rating} / big!`, () => {
      const wrapper = shallow(<Star {...star} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
