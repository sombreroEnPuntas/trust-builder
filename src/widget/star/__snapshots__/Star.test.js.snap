// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star gets proper class name and structure for: index 1 / rating: 1 / big! 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-1 star-block_size-big\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 1 / rating: 1 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-1 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 1 / rating: 2 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-2 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 1 / rating: 3 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-3 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 1 / rating: 4 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-4 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 1 / rating: 5 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-5 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;

exports[`Star gets proper class name and structure for: index 2 / rating: 1 1`] = `
"<span className=\\"Star\\">
  <span className=\\"star-block_color-0 star-block_size-small\\">
    ğŸŸŠ
  </span>
</span>"
`;
